File: src/main/java/com/guardanis/blank/BaseActivity.java
package com.guardanis.blank;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import com.guardanis.blank.tools.views.ToolbarLayoutBuilder;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    public void onCreate(Bundle savedInstance) {
        super.onCreate(savedInstance);
    }

    protected void setupToolbar(){
        setupToolbar(R.id.base__toolbar, R.layout.base__toolbar);
    }

    protected void setupToolbar(int toolbarId){
        setupToolbar(toolbarId, R.layout.base__toolbar);
    }

    protected void setupToolbar(int toolbarId, int inflatedResId){
        setup(new ToolbarLayoutBuilder((Toolbar) findViewById(toolbarId), inflatedResId));
    }

    protected abstract void setup(ToolbarLayoutBuilder builder);

}

File: src/main/java/com/guardanis/blank/BaseActivity.kt
package com.guardanis.blank

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar

import com.guardanis.blank.tools.views.ToolbarLayoutBuilder

abstract class BaseActivity: AppCompatActivity() {

    public override fun onCreate(savedInstance: Bundle?) {
        super.onCreate(savedInstance)
    }

    @JvmOverloads protected fun setupToolbar(toolbarId: Int = R.id.base__toolbar, inflatedResId: Int = R.layout.base__toolbar) {
        val toolbarView = findViewById<Toolbar>(toolbarId)
        val builder = ToolbarLayoutBuilder(toolbarView, inflatedResId)

        setup(builder)
    }

    protected abstract fun setup(builder: ToolbarLayoutBuilder)
}
